packages/extension-seo-url-mapping/src/setup-app.ts

```diff
@@ -0,0 +1,110 @@
+}
+
+// This is safe to delete if your extension does not use state. If you aren't using this, ensure you remove the
+// `withApplicationExtensionStore` usage below as well.
+const sliceInitializer: SliceInitializer<StoreSlice> = (set) => ({
+    count: 0,
+    increment: () => set((state) => ({count: state.count + 1})),
+    decrement: () => set((state) => ({count: state.count - 1}))
+})
+
+// TODO: change name of extension
+class SeoUrlMappingExtension extends ApplicationExtension<Config> {
+    static readonly id = extensionMeta.id
+
+    /**
+     * Use this method to wrap or enhance your PWA-Kit application using [React higher-order components](https://legacy.reactjs.org/docs/higher-order-components.html).
+     * You can use this to add visual treatments to your application, change the props that are supplied to the application component
+     * or add things like providers and contexts to be used throughout your app.
+     */
+    extendApp<T extends React.ComponentType<T>>(
+        App: React.ComponentType<T>
+    ): React.ComponentType<T> {
+        const HOCs = [
+            // Optionally include state for this extension using `withApplicationExtensionStore`
+            (component: React.ComponentType<any>) =>
+                withApplicationExtensionStore(component, {
+                    id: extensionMeta.id,
+                    initializer: sliceInitializer
+                })
+        ]
+
+        return applyHOCs(App, HOCs)
+    }
+
+    /**
+     * This method is used to make changes to the PWA-Kit application routes. If your extension adds a new page to the application
+     * then you can add it to the router here. The routes passed to this method is an accrued list of routes that have been added
+     * from extensions applied before it. It is called during the `getRoutes` phase on both the server and the client.
+     *
+     * NOTE: If you instead want to modify a list of all the routes, refer to the `beforeRouteMatch` below.
+     */
+    extendRoutes(routes: RouteProps[]): RouteProps[] {
+        return routes
+    }
+
+    /**
+     * This method is used on the server during the rendering pipeline. It's provided a list of all the routes that your application
+     * is configured with, including those defined in the base application and those added by all the extensions. You can use this
+     * method to modify these routes in any way you want, but you must return an array of routes as a result.
+     */
+    // TODO: make async so it can make the seo API call
+    async beforeRouteMatch(allRoutes: RouteProps[], locals: any): Promise<RouteProps[]> {
+        // TODO: Get map from config ----
+        const config = this.getConfig()
+        let {resourceTypeToComponentMap} = config
+        console.log('config', config, 'resourceTypeToComponentMap', resourceTypeToComponentMap)
+        if (resourceTypeToComponentMap === undefined) {
+            resourceTypeToComponentMap = generateResourceTypeMap(allRoutes)
```

Comment:
It seems like callback functions are not supported values for the `config/default.js`. Config fields with a callback function as the value get passed as undefined. 

As a temporary workaround, I've copied the config code to a `generateResourceTypeMap ` util function in the sdk.